### PR TITLE
jaxrs - return a HTTP 400 instead of 500 when a POST request contains invalid json

### PR DESCRIPTION
**Pull request by WhiteCat22**
https://github.com/OpenLiberty/open-liberty/issues/10986

### CODE REVIEW

**Comment by WhiteCat22** [comment]
#run-libby-bot

**Comment by WhiteCat22** [comment]
#build

File: dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/simpleJson/src/com/ibm/ws/jaxrs/fat/simpleJson/JaxrsJsonClientTestServlet.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        httpClient.getConnectionManager().shutdown();
+    }
+
+    @Test
+    public void simpleTest() throws Exception {
+        Car corvette = new Car();
+        corvette.color = "red";
+        corvette.make = "Chevrolet";
+        corvette.model = "Corvette";
+        corvette.year = 2014;
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/simpleJson/src/com/ibm/ws/jaxrs/fat/simpleJson/JaxrsJsonClientTestServlet.java]: What an oddly-specific vehicle... ;-)

File: dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/simpleJson/src/com/ibm/ws/jaxrs/fat/simpleJson/JaxrsJsonClientTestServlet.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        boolean exception = false;
+        try {
+            response.readEntity(Foo.class);
+        } catch (ResponseProcessingException e) {
+            exception = true;
+        }
+
+        if(!exception) {
+            fail();
+        }
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/simpleJson/src/com/ibm/ws/jaxrs/fat/simpleJson/JaxrsJsonClientTestServlet.java]: You could remove this and just call `fail()` in the try block after `response.readEntity(Foo.class);`

File: dev/com.ibm.ws.jaxrs.2.0_fat/publish/servers/com.ibm.ws.jaxrs.fat.simpleJson/server.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<server>
+    <featureManager>
+        <feature>componenttest-1.0</feature>
+        <feature>jaxrs-2.0</feature>
+    </featureManager>
+    
+  	<include location="../fatTestPorts.xml"/>
+  	<javaPermission className="java.io.FilePermission" name="ALL FILES" actions="read"/>
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.0_fat/publish/servers/com.ibm.ws.jaxrs.fat.simpleJson/server.xml]: Why is this necessary?  Something required by Apache HTTP client?

File: dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/JAXRSUtils.java [hunk 4]
```diff
@@ -1435,9 +1437,11 @@ public Object run() throws IOException, WebApplicationException {
                 });
             } catch (PrivilegedActionException e) {
                 Exception e1 = e.getException();
-                if (e1 instanceof IOException)
+                if (e1 instanceof JsonParseException) {
+                    throw new BadRequestException(e1);
+                } else if (e1 instanceof IOException) {
                     throw (IOException) e1;
-                else
+                } else
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/JAXRSUtils.java]: You can just remove the `else` keyword here.

File: dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/JAXRSUtils.java [hunk 5]
```diff
@@ -141,6 +142,7 @@
 import org.apache.cxf.message.MessageUtils;
 import org.apache.cxf.phase.PhaseInterceptorChain;
 import org.apache.cxf.service.Service;
+import org.codehaus.jackson.JsonParseException;
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/JAXRSUtils.java]: I'm not sure that we want to put Jackson dependencies in the JAXRSUtils class.  Is it possible to update the Jackson MBR/MBW to throw BadRequestException instead and then leave JAXRSUtils unchanged?

File: dev/com.ibm.ws.jaxrs.2.0_fat/publish/servers/com.ibm.ws.jaxrs.fat.simpleJson/server.xml (hunk 3 above)
Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.2.0_fat/publish/servers/com.ibm.ws.jaxrs.fat.simpleJson/server.xml]: it was copied by an existing server, didn't investigate

File: dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/JAXRSUtils.java (hunk 5 above)
Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/JAXRSUtils.java]: We could probably modify our wrapper class, I'd have to look into it further

File: dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/JAXRSUtils.java (hunk 4 above)
Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/JAXRSUtils.java]: I prefer else blocks for uniformity (and brackets)

File: dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/JAXRSUtils.java (hunk 5 above)
Comment (WhiteCat22) [reply dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/JAXRSUtils.java]: I was right, we pull in one of Jackson's prebuilt wrapper classes directly, we'd have to make our wrapper for their wrapper

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
#build

**Comment by WhiteCat22** [comment]
green build

**Comment by WhiteCat22** [comment]
#run-libby-bot
